### ISSUE TITLE
GraphQL: Make all data fetchers async

### ISSUE THREAD

**Issue by olim7t** [stargate/stargate#421]
With #400 the GraphQL web layer can now handle client requests asynchronously. We can extend that all the way to the persistence layer:
* change all of our data fetchers to return a `CompletionStage<...>` of whatever they were returning previously
* never block on the futures returned by the persistence layer. Instead, chain the transformations via future callbacks (`thenApply`, `whenComplete`, etc) (I attempted that previously but had to revert it in 9b6df32b).

We just have to be careful not to cross result page boundaries, because that would trigger a synchronous fetch. This can be achieved by using `currentPageRows()`, the way it's already done in `QueryFetcher`. This will only return one page of results, but that's what we want anyway.

This should only be done after #413 to avoid unnecessary conflicts.

**Comment by olim7t** [comment]
Closing this, see comments in #544.

**Comment by olim7t** [comment]
Reopening this for Stargate v2. Now that we have a clean demarcation from the persistence with `StargateBridgeClient`, this can be achieved if we use async gRPC calls.
